opensourceleg/actuators/dephy.py

```diff
@@ -0,0 +1,1739 @@
+)
+from opensourceleg.logging import LOGGER
+from opensourceleg.logging.decorators import (
+    deprecated,
+    deprecated_with_routing,
+    deprecated_with_suggestion,
+)
+from opensourceleg.math import ThermalModel
+from opensourceleg.safety import ThermalLimitException
+
+DEFAULT_POSITION_GAINS = ControlGains(kp=30, ki=0, kd=0, k=0, b=0, ff=0)
+
+DEFAULT_CURRENT_GAINS = ControlGains(kp=40, ki=400, kd=0, k=0, b=0, ff=128)
+
+DEFAULT_IMPEDANCE_GAINS = ControlGains(kp=40, ki=400, kd=0, k=200, b=400, ff=128)
+
+DEPHY_SLEEP_DURATION = 0.1
+
+RAD_PER_DEG = np.pi / 180
+RAD_PER_SEC_GYROLSB = np.pi / 180 / 32.8
+M_PER_SEC_SQUARED_ACCLSB = 9.80665 / 8192
+IMPEDANCE_A = 0.00028444
+IMPEDANCE_C = 0.0007812
+
+
+DEPHY_ACTUATOR_CONSTANTS = MOTOR_CONSTANTS(
+    MOTOR_COUNT_PER_REV=16384,
+    NM_PER_AMP=0.1133,
+    NM_PER_RAD_TO_K=((2 * np.pi / 16384) / IMPEDANCE_C * 1e3 / 0.1133),
+    NM_S_PER_RAD_TO_B=((np.pi / 180) / IMPEDANCE_A * 1e3 / 0.1133),
+    MAX_CASE_TEMPERATURE=80,
+    MAX_WINDING_TEMPERATURE=110,
+)
+
+
+def _dephy_voltage_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    dephy_actuator.stop_motor()
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+def _dephy_current_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    dephy_actuator.stop_motor()
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+def _dephy_position_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    dephy_actuator.stop_motor()
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+def _dephy_impedance_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    dephy_actuator.stop_motor()
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+DEPHY_CONTROL_MODE_CONFIGS = CONTROL_MODE_CONFIGS(
+    POSITION=ControlModeConfig(
+        entry_callback=lambda _: None,
+        exit_callback=_dephy_position_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=1000, ki=1000, kd=1000, k=0, b=0, ff=0),
+    ),
+    CURRENT=ControlModeConfig(
+        entry_callback=lambda _: None,
+        exit_callback=_dephy_current_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=80, ki=800, kd=0, k=0, b=0, ff=128),
+    ),
+    VOLTAGE=ControlModeConfig(
+        entry_callback=lambda _: None,
+        exit_callback=_dephy_voltage_mode_exit,
+        has_gains=False,
+        max_gains=None,
+    ),
+    IMPEDANCE=ControlModeConfig(
+        entry_callback=lambda _: None,
+        exit_callback=_dephy_impedance_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=80, ki=800, kd=0, k=1000, b=1000, ff=128),
+    ),
+)
+
+
+class DephyActuator(ActuatorBase, Device):
+    def __init__(
+        self,
+        tag: str = "DephyActuator",
+        firmware_version: str = "7.2.0",
+        port: str = "/dev/ttyACM0",
+        gear_ratio: float = 1.0,
+        baud_rate: int = 230400,
+        frequency: int = 500,
+        debug_level: int = 0,
+        dephy_log: bool = False,
+        offline: bool = False,
+        stop_motor_on_disconnect: bool = False,
```

Comment:
We probably want this default to be true right?

Reply:
Yes, changing default value to True.